src/dbnode/persist/fs/seek_manager.go

```diff
@@ -281,6 +314,10 @@ func (m *seekerManager) Test(id ident.ID, shard uint32, start time.Time) (bool,
 // Borrow returns a "borrowed" seeker which the caller has exclusive access to
 // until it's returned later.
 func (m *seekerManager) Borrow(shard uint32, start time.Time) (ConcurrentDataFileSetSeeker, error) {
+	if !m.isShardAvailable(shard) {
+		return nil, errShardNotAvailable
+	}
```

Comment:
Hm yeah, so I don't think we actually need to check the shard set in the read path - I believe we should probably just rely on the open close loop to eventually remove the shards.